moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1166,24 +1166,26 @@ void TrajectoryExecutionManager::stopExecution(bool auto_clear)
 
       // we set the status here; executePart() will not set status when execution_complete_ is true ahead of time
       last_execution_status_ = moveit_controller_manager::ExecutionStatus::PREEMPTED;
-      execution_state_mutex_.unlock();
-      ROS_INFO_NAMED(name_, "Stopped trajectory execution.");
-
       // wait for the execution thread to finish
       execution_thread_->join();
       execution_thread_.reset();
 
+      lock.unlock();
```

Comment:
Misusing `execution_state_mutex_` here to protect `execution_thread_` as well doesn't work, because the execution thread tries to lock the mutex as well to indicate that it's done via the `execution_complete_` flag.